src/components/buttons/accessible-button/accessible-button.js

```diff
@@ -67,7 +68,23 @@ AccessibleButton.displayName = 'AccessibleButton';
 AccessibleButton.propTypes = {
   as: PropTypes.oneOfType([PropTypes.string, PropTypes.elementType]),
   id: PropTypes.string,
-  type: PropTypes.oneOf(['submit', 'reset', 'button']),
+  type: (props, propName, componentName, ...rest) => {
+    // the type defaults to `button`, so we don't need to handle undefined
+    if (props.as && props.type !== 'button') {
```

Comment:
we can keep this here because `AccessibleButton` only knows about `as` and not `linkTo`